tensorboard/webapp/core/views/hash_storage_test.ts

```diff
@@ -70,6 +70,17 @@ describe('hash storage test', () => {
     fixture.detectChanges();
     expect(setPluginIdSpy).toHaveBeenCalledWith('foo', {
       useLocationReplace: true,
+      defaultValue: '',
+    });
+  });
+
+  it('sets the hash to empty string when activePlugin is not set', () => {
```

Comment:
You might to add a test for `getActivePlugin()` returning empty string. Not sure if that's a possible scenario.

Reply:
That is not a possible scenario. plugin id is a required field (although neither python nor JS part checks it) and expected to be non-empty.

Reply:
In the hindsight, it is technically possible to have the empty string when the store bootstraps from the empty hash (`/#`). Added the case.